src/meta/src/barrier/mod.rs

```diff
@@ -782,15 +593,18 @@ impl GlobalBarrierManager {
             checkpoint,
             span,
         } = self.scheduled_barriers.pop_or_default().await;
-        self.checkpoint_control.pre_resolve(&command);
-        let info = self
+
+        let all_nodes = self
             .context
-            .resolve_actor_info(|s: ActorState, table_id: TableId, actor_id: ActorId| {
-                self.checkpoint_control
-                    .can_actor_send_or_collect(s, table_id, actor_id)
-            })
-            .await;
-        self.checkpoint_control.post_resolve(&command);
+            .metadata_manager
+            .list_active_streaming_compute_nodes()
+            .await
+            .unwrap();
+        self.state.resolve_worker_nodes(all_nodes);
+
+        self.state.pre_resolve(&command);
```

Comment:
Can we merge the 3 lines into `let info = self.state.apply_command(&command)`, which atomically applies the command to the state and return the actor info corresponding to the command, and then we may not need these 3 separate methods.